plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StartTask.java

```diff
@@ -0,0 +1,354 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.gradle.BuildListener;
+import org.gradle.BuildResult;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository;
+import org.gradle.api.initialization.Settings;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.bundling.Jar;
+import org.gradle.api.tasks.bundling.War;
+import org.gradle.initialization.BuildCancellationToken;
+import org.wildfly.swarm.bootstrap.util.BootstrapProperties;
+import org.wildfly.swarm.fractions.FractionDescriptor;
+import org.wildfly.swarm.fractions.FractionList;
+import org.wildfly.swarm.fractions.FractionUsageAnalyzer;
+import org.wildfly.swarm.tools.ArtifactResolvingHelper;
+import org.wildfly.swarm.tools.ArtifactSpec;
+import org.wildfly.swarm.tools.BuildTool;
+import org.wildfly.swarm.tools.DependencyManager;
+import org.wildfly.swarm.tools.exec.SwarmExecutor;
+import org.wildfly.swarm.tools.exec.SwarmProcess;
+
+public class StartTask extends DefaultTask {
+
+    private static final String EXCLUDE_PREFIX = "!";
+    private static final String INDENTATION_DELIMITER = "\n  ";
+
+    private ThorntailExtension extension;
+    private boolean waitForProcess = false;
+
+    public StartTask() {
+        extension = getProject().getExtensions().getByType(ThorntailExtension.class);
+    }
+
+    void waitForProcess() {
+        waitForProcess = true;
+    }
+
+    @TaskAction
+    public void startApplication() {
+        final SwarmExecutor executor;
+        if (extension.isUseUberJar()) {
+            executor = uberJarExecutor();
+        } else if (!getProject().getTasks().withType(War.class).isEmpty()) {
+            executor = warExecutor();
+        } else if (!getProject().getTasks().withType(Jar.class).isEmpty()) {
+            executor = jarExecutor();
+        } else {
+            throw new GradleException("Unable to determine type of the project, neither war or jar tasks were found.");
+        }
+
+        executor.withJVMArguments(extension.getJvmArguments());
+        executor.withArguments(extension.getArguments());
+
+        final SwarmProcess process;
+        try {
+
+            File tmp;
+            try {
+                tmp = Files.createTempFile("thorntail-process-file", null).toFile();
+                getLogger().info("Process file: " + tmp.toPath().toString());
+            } catch (IOException e) {
+                throw new GradleException("Error while creating process file");
+            }
+
+            process = executor.withDebug(extension.getDebugPort())
```

Comment:
That's a good idea, I didn't thought of that.

What I thought about and then forgot was to enable for the debug port to be overridden by a property from a command line, something like adding "-Pthorntail.debugPort=8888" to the gradle command so that we don't have to modify the build file to enable debugging. Maybe would leave that also for a latter PR, not to complicate this one further.

Reply:
Only speaking about the proposed system property name: there's a property used for setting a debug port already: `thorntail.debug.port`. I think it would be good to stick to existing names.

Reply:
Agreed.

Reply:
Good point, it will look for a project property "thorntail.debug.port".

Reply:
* Detecting "thorntail.debug.port" project property is done.
* Leaving check that port is free for later.